File: extractor/treesitter/testdata/sample.py
"""Sample Python module for testing."""

import os
from pathlib import Path


class Greeter:
    """A class that greets people."""

    def greet(self, name: str) -> str:
        """Return a greeting for name."""
        return f"Hello, {name}!"


def main():
    """Entry point."""
    g = Greeter()
    print(g.greet("world"))


# A constant
DEFAULT_TIMEOUT = 30

File: extractor/treesitter/testdata/sample.ts
// Sample TypeScript module for testing.

import { readFileSync } from "fs";

export interface Greeter {
  greet(name: string): string;
}

export class SimpleGreeter implements Greeter {
  greet(name: string): string {
    return `Hello, ${name}!`;
  }
}

export function main(): void {
  const g = new SimpleGreeter();
  console.log(g.greet("world"));
}

type Config = {
  port: number;
  host: string;
};

export enum LogLevel {
  DEBUG,
  INFO,
  WARN,
  ERROR,
}
